pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -146,6 +152,27 @@ func (r *TokenREST) Create(ctx context.Context, name string, obj runtime.Object,
 				return nil, errors.NewBadRequest(fmt.Sprintf("cannot bind token for serviceaccount %q to pod running with different serviceaccount name.", name))
 			}
 			uid = pod.UID
+			if utilfeature.DefaultFeatureGate.Enabled(features.ServiceAccountTokenPodNodeInfo) {
+				if nodeName := pod.Spec.NodeName; nodeName != "" {
+					newCtx := newContext(ctx, "nodes", nodeName, api.SchemeGroupVersion.WithKind("Node"))
+					nodeObj, err := r.nodes.Get(newCtx, nodeName, &metav1.GetOptions{})
```

Comment:
yes, though not if the feature gate check above is passing, so this shouldn't ever happen. `r.pods` *could* also be nil based solely on the NewREST method signature - do we want extra-extra safe checks here to ensure `r.nodes` is not nil as well as the feature gate being enabled? I could adapt this to return an error if the gate is enabled and `r.nodes` is nil as a defence in depth against crashes?

Reply:
Maybe? I don't really understand how the generic construction path is used.

@sttts, is https://github.com/kubernetes/kubernetes/pull/119082/files#r1365556460 enabling serving service accounts and handling TokenReview API requests without pod and node storage existing? If so, work is required to make that return API errors inside this storage handler instead of panicing (and means pod and node bindings in the TokenReview API would error when submitted to those servers).

Reply:
this was improved in https://github.com/kubernetes/kubernetes/pull/121372 to plumb placeholder NotFound getters in the generic construction path. Rebase this and add a NotFound getter for the node getter as well